docs/dev/design/better-doc-urls-handling.rst

```diff
@@ -0,0 +1,417 @@
+     - 404
+     - subproject
+     - The subproject is single version
+   * - /projects/subproject/404
+     - 404
+     - subproject
+     - The subproject exists, but the file doesn't
+   * - /projects/subproject/br/latest/manual/index.html
+     - /latest/br/latest/manual/index.html
+     - subproject
+     - The subproject has a ``br`` directory!
+   * - /projects/nothing/manual/index.html
+     - 404
+     - project
+     - The subproject doesn't exist
+   * - /en/latest/manual/index.html
+     - /latest/manual/index.html
+     - project
+     -
+   * - /404
+     - 404
+     - project
+     -
+
+Project with custom prefix
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- project (latest, 1.0)
+- subproject (latest, 1.0)
+
+``project`` has the ``prefix`` prefix, and ``sub`` subproject prefix.
+
+.. list-table:: 
+   :header-rows: 1
+
+   * - Request
+     - Requested file
+     - Current project
+     - Note
+   * - /en/latest/manual/index.html
+     - 404
+     - project
+     - The prefix doesn't match
+   * - /prefix/en/latest/manual/index.html
+     - /latest/manual/index.html
+     - project
+     -
+   * - /projects/subproject/en/latest/manual/index.html
+     - 404
+     - project
+     - The subproject prefix doesn't match
+   * - /sub/subproject/en/latest/manual/index.html
+     - /latest/manual/index.html
+     - subproject
+     -
+   * - /sub/nothing/en/latest/manual/index.html
+     - 404
+     - project
+     - The subproject doesn't exist
+
+Project with custom subproject prefix (empty)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- project (latest, 1.0)
+- subproject (latest, 1.0)
+
+``project`` has the ``/`` subproject prefix,
+this allow us to serve subprojects without using a prefix.
+
+.. list-table:: 
+   :header-rows: 1
+
+   * - Request
+     - Requested file
+     - Current project
+     - Note
+   * - /en/latest/manual/index.html
+     - /latest/manual/index.html
+     - project
+     -
+   * - /projects/subproject/en/latest/manual/index.html
+     - 404
+     - project
+     - The subproject prefix doesn't match
+   * - /subproject/en/latest/manual/index.html
+     - /latest/manual/index.html
+     - subproject
+     -
+   * - /nothing/en/latest/manual/index.html
+     - /latest/manual/index.html
+     - project
+     - The subproject/file doesn't exist
+
+Questions
+---------
+
+- When using custom URLs,
+  should we support changing the URLs
+  that aren't related to doc serving?
+
+  These are:
+
+  - Health check
+  - Proxied APIs
+  - robots and sitemap
+  - The ``page`` redirect
+
+  I'd say we shouldn't, I can't think of a reason why this should be supported.
+
+- Should we use the urlconf from the subproject when processing it?
+  This is an URL like ``/projects/subproject/custom/prefix/en/latest/index.html``.
+
+  I don't think that's useful, but it should be easy to support if needed.
+
+- Should we support the page redirect when using a custom subproject prefix?
+  This is ``/{prefix}/subproject/page/index.html``.
```

Comment:
Not a requirement, but would be useful if it isn't difficult. Would also be cool if users could customize it :)

`/{prefix}/subproject/yolo-latest-version/index.html` 🚀